File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/CBuilder/UnboundModeDemo/UnboundModeDemoCustomField.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "UnboundModeDemoCustomField.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxButtons"
#pragma link "cxLookAndFeelPainters"
#pragma resource "*.dfm"
TUnboundModeDemoCustomFieldForm *UnboundModeDemoCustomFieldForm;
//---------------------------------------------------------------------------
__fastcall TUnboundModeDemoCustomFieldForm::TUnboundModeDemoCustomFieldForm(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------

int __fastcall TUnboundModeDemoCustomFieldForm::ShowModal()
{
  SetFormPosition(this, 35, 45);
  return (TForm::ShowModal());
}
//---------------------------------------------------------------------------

void __fastcall TUnboundModeDemoCustomFieldForm::edtKeyPress(
      TObject *Sender, char &Key)
{
  if ((Key < '0') || ('9' < Key))
    Key = 7;
}
//---------------------------------------------------------------------------

File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/Delphi/UnboundModeDemo/UnboundModeDemoCustomField.pas
unit UnboundModeDemoCustomField;

interface

uses
{$IFDEF CLR}
  System.ComponentModel,
{$ENDIF}
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls, cxButtons, cxLookAndFeelPainters;

type
  TUnboundModeDemoCustomFieldForm = class(TForm)
    lbHeight: TLabel;
    lbWidth: TLabel;
    lbMineCount: TLabel;
    edtHeight: TEdit;
    edtWidth: TEdit;
    edtMineCount: TEdit;
    btnOK: TcxButton;
    bntCancel: TcxButton;
    procedure edtKeyPress(Sender: TObject; var Key: Char);
  public
    function ShowModal: Integer; override;
  end;

implementation

uses UnboundModeDemoTypes;

{$R *.DFM}

function TUnboundModeDemoCustomFieldForm.ShowModal: Integer;
begin
  SetFormPosition(Self, 25, 25);
  Result := inherited ShowModal;
end;

procedure TUnboundModeDemoCustomFieldForm.edtKeyPress(Sender: TObject;
  var Key: Char);
begin
  if (Key < '0') or ('9' < Key) then
    Key := Char(7);
end;

end.
